vignettes/ospsuite.Rmd

```diff
@@ -11,96 +11,339 @@ vignette: >
-* [Table parameters](table-parameters.html)
-* [Dimensions and Units](unit-conversion.html)
-* [Working with data sets and import from excel](observed-data.html)
-* [Working with `DataCombined` class](data-combined.html)
-* [Visualizations with `DataCombined`](data-combined-plotting.html)
+```r
+system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+```
+
+
+## Quick Start: Basic Simulation
+
+This example demonstrates the essential ospsuite workflow. Detailed explanations follow in subsequent sections.
+
+```{r, echo=TRUE, results='hide'}
+# Load the ospsuite package
+library(ospsuite)
+
+# Load the built-in example simulation
+simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+sim <- loadSimulation(simFilePath)
+
+# Run the simulation
+results <- runSimulations(sim)
+
+# Create a quick visualization
+myDataCombined <- DataCombined$new()
+myDataCombined$addSimulationResults(results[[1]])
+plotIndividualTimeProfile(dataCombined = myDataCombined)
+```
+
+This example loads, runs, and visualizes a PBPK simulation. The plot shows the concentration-time profile for Aciclovir in plasma after IV administration.
+
+## Step-by-Step Tutorial
+
+The following sections detail each workflow step and demonstrate ospsuite capabilities.
+
+### Loading Simulations
+
+Simulations require a **.pkml** format file (exported from PK-Sim or MoBi):
+
+```{r, echo=TRUE, message=TRUE}
+# Using the included example
+simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+sim <- loadSimulation(simFilePath)
+
+# For custom files, use:
+# sim <- loadSimulation("path/to/simulation.pkml")
+
+# Explore the simulation
+print(sim)
+```
+
+### Exploring Model Structure
+
+After loading, explore the simulation structure and parameters:
+
+```{r, echo=TRUE}
+# Get the simulation tree structure
+simTree <- getSimulationTree(sim)
+
+# Explore parameter paths
+simTree$Organism$Weight$path
+
+# Find parameters using wildcards
+getAllParametersMatching("**|Dose*", sim)
+
+# Or search for specific terms
+grep("Dose", getAllParameterPathsIn(container = sim), value = TRUE)
+```
+
+**Note:** Parameter paths in ospsuite match those displayed in PK-Sim or MoBi.
+
+### Modifying Parameters
+
+Retrieve and modify parameter values:
+
+```{r, echo=TRUE, message=TRUE}
+# Get a specific parameter
+dose <- getParameter(
+  path = "Applications|IV 250mg 10min|Application_1|ProtocolSchemaItem|Dose",
+  sim
+)
+print(dose)
+
+# Change the dose value
+setParameterValues(dose, 0.004) # New dose: 4 mg/kg
+print(dose)
+
+# Scale a parameter by a factor
+scaleParameterValues(dose, factor = 2) # Double the dose
+print(dose)
+
+# Reset to original value
+dose$reset()
+print(dose)
+```
+
+### Customizing Simulation Settings
+
+Adjust simulation outputs and solver settings:
+
+```{r, echo=TRUE, message=TRUE}
+# Add new outputs to track
+addOutputs(c("Organism|Lumen|Stomach|Aciclovir"), simulation = sim)
+
+# Adjust solver precision if needed
```

Comment:
In this example, I would add `sim$outputSelections` **before** and **after** `addOutputs` and explain that "`Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)`" was already included as an output in the loaded pkml.

Otherwise, it's unclear where the _Plasma (Peripheral Venous Blood)_ comes from, when using it in all further examples below